flatfs.go

```diff
@@ -94,10 +168,25 @@ func Open(path string, sync bool) (*Datastore, error) {
 	}
 
 	fs := &Datastore{
-		path:     path,
-		shardStr: shardId.String(),
-		getDir:   shardId.Func(),
-		sync:     sync,
+		path:      path,
+		shardStr:  shardId.String(),
+		getDir:    shardId.Func(),
+		sync:      syncFiles,
+		diskUsage: 0,
+		opMap: &opMap{
+			ops: sync.Map{},
```

Comment:
No need to manually initialize this. `opMap: new(opMap)` is sufficient. Same with `diskUsage` (go initializes to "zero value").